examples/topics/categorical/correlogram.py

```diff
@@ -0,0 +1,73 @@
+''' A categorical plot showing the correlations in mineral content for 214 samples of glass fragments
+obtained during forensic work.
+
+The dataset contains seven variables measuring the amounts of magnesium (Mg),
+calcium (Ca), iron (Fe), potassium (K), sodium (Na), aluminum (Al), and barium (Ba) found in each glass fragment.
+The colored circles represent the correlations between pairs of these variables.
+The magnitude of each correlation is encoded in the size of the circles.
+
+.. bokeh-example-metadata::
+    :sampledata: forensic_glass
+    :apis: bokeh.plotting.figure.scatter
+    :refs: :ref:`ug_topics_categorical_scatters`
+    :keywords: scatter, correlogram
+
+'''
+
+from itertools import combinations
+
+import numpy as np
+import pandas as pd
+
+from bokeh.models import ColorBar, ColumnDataSource, FixedTicker, LinearColorMapper
+from bokeh.plotting import figure, show
+from bokeh.sampledata.forensic_glass import data as df
+from bokeh.transform import transform
+
+elements = ("Mg", "Ca", "Fe", "K", "Na", "Al", "Ba")
+
+pairs = list(combinations(elements, 2))
+x, y = list(zip(*pairs))
+
+correlations = []
+for x, y in pairs:
+    matrix = np.corrcoef(df[x], df[y])
+    correlations.append(matrix[0, 1])
+
+new_df = pd.DataFrame({
+    "oxide_1": [x[0] for x in pairs],
+    "oxide_2": [x[1] for x in pairs],
+    "correlation": correlations,
+    "dot_size": [(1+ 10 * abs(corr)) * 10 for corr in correlations],
+})
+
+x_range = new_df["oxide_1"].unique()
+y_range = list(new_df["oxide_2"].unique())
+
+source = ColumnDataSource(new_df)
+
+mapper = LinearColorMapper(palette="RdYlGn9", low=-0.5, high=0.5)
+
+p = figure(x_axis_location="above", toolbar_location=None, x_range=x_range, y_range=y_range, background_fill_color="#fafafa")
+
+p.scatter(x="oxide_1", y="oxide_2", size="dot_size", source=source, fill_color=transform("correlation", mapper), line_color="#202020")
```

Comment:
I suppose this is OK but FYI typically we try to demonstrate the convenience function `linear_cmap` which avoids having to use both `transform` and `LinearColorMapper` directly.

Reply:
The `color_bar` variable also uses the `LinearColorMapper` as well to build the `ColorBar`. I'm not sure what to use as the `color_mapper` then if I remove the `LinearColorMapper` altogether.

Reply:
The solution is to use the `construct_color_bar` function rather than create a `ColorBar` explicitly. After making Bryan's recommended change, store the object returned by the `scatter` call, e.g.
```python
s = p.scatter(x="oxide_1", ...)
```
then replace the `color_bar = ColorBar` call with
```python
color_bar = s.construct_color_bar(location= ...)
```
The arguments that you pass to `construct_color_bar` as the same as you have for `ColorBar` but without the `color_mapper` as this automatically read from `s`.

Reply:
Thanks.